File: java/347. Top K Frequent Elements.java
public class Solution {
    public List<Integer> topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> map1=new HashMap<Integer, Integer>();
        Map<Integer,List<Integer>> map2=new HashMap<Integer, List<Integer>>();
        for (int i = 0; i < nums.length; i++) {
            map1.put(nums[i], map1.get(nums[i])==null?1:(map1.get(nums[i])+1));
        }
        Set<Integer> set=map1.keySet();
        int[] arrays=new int[set.size()];
        int a=0;
        for (Integer key : set) {
            int value=map1.get(key);
            if (map2.get(value)==null) {
                List<Integer> list=new ArrayList<Integer>();
                list.add(key);
                map2.put(value, list);
                arrays[a++]=value;
            }else {
                map2.get(value).add(key);
            }
        }
        Arrays.sort(arrays);
        List<Integer> list=new ArrayList<Integer>();
        a=arrays.length-1;
        while (list.size()<k) {
            list.addAll(map2.get(arrays[a--]));
        }
        return list;
    }
}

File: python/347. Top K Frequent Elements.py
#coding=utf-8
import collections

__author__ = 'xuxuan'
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        map1={}
        map2={}
        for num in nums:
            map1[num]=map1.get(num,0)+1
        for key,value in map1.iteritems():
            map2[value]=map2.get(value,[])+[key]
        arrays=map2.keys()
        arrays.sort(reverse=True)
        ans=[]
        a=0
        while len(ans)<k:
            ans+=map2.get(arrays[a])
            a+=1
        return ans

    #这是一句话的方法
    def oneSentenceMethod(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        return [x[0] for x in collections.Counter(nums).most_common(k)]